collect_app/src/main/java/org/odk/collect/android/activities/FillBlankFormActivity.kt

```diff
@@ -0,0 +1,351 @@
+            val progressBar = findViewById<ObviousProgressBar>(R.id.progressBar)
+            if (syncing) {
+                progressBar.show()
+            } else {
+                progressBar.hide(View.GONE)
+            }
+        }
+
+        blankFormsListViewModel.isAuthenticationRequired.observe(this) { authenticationRequired ->
+            if (authenticationRequired) {
+                showIfNotShowing(ServerAuthDialogFragment::class.java, supportFragmentManager)
+            } else {
+                dismissDialog(ServerAuthDialogFragment::class.java, supportFragmentManager)
+            }
+        }
+
+        menuDelegate =
+            BlankFormListMenuDelegate(this, blankFormsListViewModel, networkStateProvider)
+
+        init()
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        val result = super.onCreateOptionsMenu(menu)
+        menuDelegate.onCreateOptionsMenu(menuInflater, menu)
+        return result
+    }
+
+    override fun onPrepareOptionsMenu(menu: Menu): Boolean {
+        val result = super.onPrepareOptionsMenu(menu)
+        menuDelegate.onPrepareOptionsMenu(menu)
+        return result
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        if (!allowClick(javaClass.name)) {
+            return true
+        }
+        return if (super.onOptionsItemSelected(item)) {
+            true
+        } else {
+            menuDelegate.onOptionsItemSelected(item)
+        }
+    }
+
+    private fun init() {
+        setupAdapter()
+
+        // DiskSyncTask checks the disk for any forms not already in the content provider
+        // that is, put here by dragging and dropping onto the SDCard
+        formSyncTask = lastCustomNonConfigurationInstance as FormSyncTask?
+        if (formSyncTask == null) {
+            Timber.i("Starting new disk sync task")
+            formSyncTask = FormSyncTask()
+            formSyncTask!!.setDiskSyncListener(this)
+            formSyncTask!!.execute()
+        }
+        sortingOptions = intArrayOf(
+            R.string.sort_by_name_asc, R.string.sort_by_name_desc,
+            R.string.sort_by_date_asc, R.string.sort_by_date_desc
+        )
+        setupAdapter()
+        supportLoaderManager.initLoader(LOADER_ID, null, this)
+    }
+
+    override fun onRetainCustomNonConfigurationInstance(): Any? {
+        // pass the thread on restart
+        return formSyncTask
+    }
+
+    /**
+     * Stores the path of selected form and finishes.
+     */
+    override fun onItemClick(parent: AdapterView<*>?, view: View, position: Int, id: Long) {
+        if (allowClick(javaClass.name)) {
+            // get uri to form
+            val idFormsTable = listView.adapter.getItemId(position)
+            val formUri = FormsContract.getUri(
+                currentProjectProvider.getCurrentProject().uuid,
+                idFormsTable
+            )
+            val action = intent.action
```

Comment:
There is no need to create this additional object. I would call 
`Intent.ACTION_PICK == intent.action`
below directly.

Reply:
Ah, yeah definitely better to inline this. Just for clarity though: `val action = intent.action` won't create a new object, it just creates a new reference to an existing object (the object returned by `intent.action`).